### PR TITLE
Tidied header files and reduced dependencies

### PR DESCRIPTION
**Pull request by martinbudden**
Reduced dependencies by moving some #includes out of header files into .c files.

### CODE REVIEW

File: src/main/drivers/compass_hmc5883l.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 #pragma once
 
-#include "drivers/io.h"
```
Comment (ledvinap) [inline src/main/drivers/compass_hmc5883l.h]: Is there any advantage to removing directory names? With directory prefix, it is easier to find the file ...

Comment (martinbudden) [inline src/main/drivers/compass_hmc5883l.h]: It's just for consistency. Some driver headers have the directory prefix, some don't. I don't have a preference for either, but it is good for them to all be the same.

Comment (ledvinap) [inline src/main/drivers/compass_hmc5883l.h]: My vote is to use the directory prefix, possibly with exception of '.h' matching '.c' file.

We can also modify Makefile to restrict `-I` directories as much as practical ...

Comment (martinbudden) [inline src/main/drivers/compass_hmc5883l.h]: Well there's lot's of header file cleanup required, so we can add the directory prefix in a subsequent PR.

Comment (unitware) [inline src/main/drivers/compass_hmc5883l.h]: I think that every step to a better state is worth taking, but if we want the includes to include the directory prefix then some of this PR is going the other way. So I suggest that we at least do not make ot worse by removing the directory prefixes in this PR. - That said we do not need to add missing prefixes in all includes.

**Comment by martinbudden** [comment]
@hydra , can this be merged in now?

**Comment by hydra** [comment]
@martinbudden can we keep the directory prefixes? then I'll merge it.

**Comment by martinbudden** [comment]
@hydra , the vast majority of the `#includes` in the `drivers` directory do not use the directory prefix. As I said before if we want to change that and make it consistent then there should be a single PR to do so.

Note that by doing that it will make merging in betaflight and iNav drivers more difficult.

**Comment by hydra** [comment]
@martinbudden ok, i will merge as is then.

**Comment by hydra** [comment]
@martinbudden actually this needs a rebase due to the motors/servos split.  Can you update it please?

**Comment by martinbudden** [comment]
Rebased

**Comment by hydra** [comment]
great!

**Pull request merged**